src/rocdata.jl

```diff
@@ -22,90 +23,91 @@ function _vector2labels(labels, truelabel)
     end
 end
 
-function _preparedrocdata(scores, labels, reverseordered::Bool)
+function _preparedrocdata(scores, labels, distancescored)
 	if length(scores) == length(labels)
-		_create_preparedrocdata(scores,labels,reverseordered)
+		_create_preparedrocdata(scores,labels,distancescored)
 	else
 		error("scores and labels should have the same length")
 	end
 end
 
 struct ROCData{T<:Real}
-	scores::Vector{T}
-	labels::BitVector
+	scores::AbstractVector{T}
+	labels::AbstractVector{Bool}
 	P::Int
 	n::Int
 	N::Int
 	ni::UnitRange{Int}
-	TP::Vector{Int}
-	TN::Vector{Int}
-	FP::Vector{Int}
-	FN::Vector{Int}
-	FPR::Vector{Float64}
-	TPR::Vector{Float64}
+	TP::AbstractVector{Int}
+	TN::AbstractVector{Int}
+	FP::AbstractVector{Int}
+	FN::AbstractVector{Int}
+	FPR::AbstractVector{Float64}
+	TPR::AbstractVector{Float64}
 end
 
 function roc(_preparedrocdata::_PreparedROCData)
 	P = sum(_preparedrocdata.labels)
-	n = length(_preparedrocdata.labels)
+	n = length(_preparedrocdata.labels)+1
```

Comment:
I don't understand this. `n`, i.e. the number of elements, should be the `length`. Otherwise, `N`, i.e. the number of negative scores, will be misleading.

Reply:
The roc threshold process needs to start before the first sample, and end after, hence the length should be length of samples + 1. I think the way to check this is to make a very small sample (say 5 elements) and check that their ROC curve looks right, and AUC is exactly correct

Reply:
I agree. I would be great to have such an example in tests to check the correctness of the approach. Also we can use `n` as `length` and `push!` the last correct value for threshold _Inf_. All this makes me realise that maybe we are going to get incorrect results with repeated thresholds having different labels. Maybe we should add a threshold field, storing the unique values of the sorted scores field but ending with an extra value (`typemax`).

Reply:
As already discussed, example given in #7